cli/cmd/create.go

```diff
@@ -48,259 +33,69 @@ var createCmd = &cobra.Command{
-		if err != nil {
-			return err
-		}
-	}
-
-	// check if image exists unless asked not to
-	if !skipCheckTag {
-		exists, err := ab.CheckImageExists(ctx, tag)
-		if err != nil {
-			return err
-		}
-		if !exists {
-			return errors.Errorf("image tag does not exist: '%s'", tag)
-		}
-	}
-
-	options := stackservice.NewStackManagementOptions(stackName).WithHappyConfig(happyConfig).WithStackService(stackService).WithBackend(backend)
-	// if we already have a stack and "force" then use existing
-	var stackMeta *stackservice.StackMeta
-	if force && existingStack != nil {
-		logrus.Infof("stack '%s' already exists, it will be updated", stackName)
-		if err != nil {
-			return err
-		}
-		options = options.WithStack(existingStack)
-		return updateStack(ctx, options)
-	} else {
-		stackMeta = stackService.NewStackMeta(stackName)
-	}
-
-	options = options.WithStackMeta(stackMeta)
-
-	// now that we have images, create all TFE related resources
-	return createStack(ctx, cmd, options)
-}
-
-func createWorkspaceRepo(isDryRun util.DryRunType, backend *backend.Backend) workspace_repo.WorkspaceRepoIface {
-	var workspaceRepo workspace_repo.WorkspaceRepoIface
-	if util.IsLocalstackMode() {
-		workspaceRepo = workspace_repo.NewLocalWorkspaceRepo().WithDryRun(isDryRun)
-	} else {
-		url := backend.Conf().GetTfeUrl()
-		org := backend.Conf().GetTfeOrg()
-		workspaceRepo = workspace_repo.NewWorkspaceRepo(url, org).WithDryRun(isDryRun)
-	}
-	return workspaceRepo
-}
-
-func createStack(ctx context.Context, cmd *cobra.Command, options *stackservice.StackManagementOptions) error {
-	var errs *multierror.Error
-	isDryRun := util.DryRunType(dryRun)
-
-	if options.Stack != nil {
-		errs = multierror.Append(errs, errors.New("stack option not expected in this context"))
-	}
-	if options.StackService == nil {
-		errs = multierror.Append(errs, errors.New("stackService option not provided"))
-	}
-	if options.Backend == nil {
-		errs = multierror.Append(errs, errors.New("backend option not provided"))
-	}
-	if options.StackMeta == nil {
-		errs = multierror.Append(errs, errors.New("stackMeta option not provided"))
-	}
-	if len(options.StackName) == 0 {
-		errs = multierror.Append(errs, errors.New("stackName option not provided"))
-	}
-
-	err := errs.ErrorOrNil()
-	if err != nil {
-		return err
-	}
-
-	secretId := options.HappyConfig.GetSecretId()
-
-	metaTag := map[string]string{"happy/meta/configsecret": secretId}
-	err = options.StackMeta.Load(metaTag)
-	if err != nil {
-		return errors.Wrap(err, "failed to load stack meta")
-	}
-
-	targetBaseTag := tag
-	if sliceDefaultTag != "" {
-		targetBaseTag = sliceDefaultTag
-	}
-
-	err = options.StackMeta.Update(ctx, targetBaseTag, options.StackTags, "", options.StackService)
-	if err != nil {
-		return errors.Wrap(err, "failed to update the stack meta")
-	}
-
-	stack, err := options.StackService.Add(ctx, options.StackName, isDryRun)
+	stackName := args[0]
+	err = validateHappyEnvironment(
+		ctx,
+		happyConfig,
```

Comment:
Same as above goes for the input parameters -- if a number of them are used together, they should be aggregated.